x-pack/legacy/plugins/reporting/types.d.ts

```diff
@@ -29,9 +29,11 @@ export interface ReportingPlugin {
   };
   // TODO: convert exportTypesRegistry to TS
   exportTypesRegistry: {
-    getById: (id: string) => ExportTypeDefinition;
-    getAll: () => ExportTypeDefinition[];
-    get: (callback: (item: ExportTypeDefinition) => boolean) => ExportTypeDefinition;
+    getById: <T, U, V, W>(id: string) => ExportTypeDefinition<T, U, V, W>;
```

Comment:
TODO: fix the generic param namings

Reply:
not needed here though - this will be removed when ExportTypesRegistry is typescripted